mysite/lti/views.py

```diff
@@ -80,20 +88,29 @@ def lti_redirect(request, course_id=None, unit_id=None):
     |  and/or UserSocialAuth.
     |  Finally login Django user and redirect to Course
 
-    :param course_id: course id from launch url
     :param unit_id: unit id from lunch url
     """
     request_dict = pickle.loads(request.session['LTI_POST'])
-    consumer_name = request_dict.get('ext_lms', 'lti')
+
+    context_id = request_dict.get('context_id')
+    course_ref = CourseRef.objects.filter(context_id=context_id).first()
+    consumer_name = request_dict.get('tool_consumer_info_product_family_code', 'lti')
     user_id = request_dict.get('user_id', None)
-    roles = ROLES_MAP.get(request_dict.get('roles', None), 'student')
-    if not user_id or not course_id:
-        return render_to_response('lti/error.html', RequestContext(request))
-    course_id = int(course_id)
-
-    user, created = LTIUser.objects.get_or_create(user_id=user_id,
-                                                  consumer=consumer_name,
-                                                  course_id=course_id)
+    roles_from_request = request_dict.get('roles', '').split(',')
+    roles = list(set((ROLES_MAP.get(role, Role.ENROLLED) for role in roles_from_request)))
+
+    if not user_id:
+        return render_to_response(
+            'lti/error.html',
+            {'message': 'There is not user_id required LTI param'},
+            RequestContext(request)
+        )
+
+    user, created = LTIUser.objects.get_or_create(
+        user_id=user_id,
+        consumer=consumer_name,
+        course_id=request_dict.get('context_id')
```

Comment:
It's a bit confusing that this LTIUser field is called course_id but is actually not a course_id but instead a context_id.  Seems like we should rename it.